develop/smart-contracts/overview.md

```diff
@@ -1,4 +1,145 @@
+
+This guide highlights some differences between building smart contract applications or parachains within Polkadot's scalable and interoperable framework. It also explores the different development paths for smart contract developers on Polkadot.
+
+!!!info "Parachain Developer?" 
+    If you are a Parachain developer looking to add smart contract functionality to your chain, please check the following sections for [WASM-based](TODO:update-path){target=\_blank} or [EVM-based contracts](TODO:update-path){target=\_blank}.
+
+## Smart Contracts Versus Parachains
+
+In a smart contract, you create a program that executes specific logic isolated to the chain on which it is being executed. All the logic executed is bound to the same state transition rules given by the underlying virtual machine (VM). Consequently, smart contracts are more streamlined to develop, and programs can easily interact with each other through similar interfaces.
+
+``` mermaid
+flowchart LR
+    A[(Chain State)] --> B[[Virtual Machine]]
+    C[Transaction] --> B
+    D["Program Logic and Storage<br/>(Smart Contract)"] --> B
+    B --> E[(New State)]
+    B --> F[Execution Logs]
+```
+
+In addition, because smart contracts are programs that execute on top of existing chains, teams don't have to think about the underlying consensus they are built on. 
+
+However, these strengths come with certain limitations. Contracts are immutable by default. Hence, developers have developed different [proxy strategies](https://blog.openzeppelin.com/proxy-patterns){target=\_blank} to be able to upgrade them over time. Generally speaking, such patterns rely on a proxy contract, where the program storage lives, forwarding a call to an implementation contract, and where the execution logic resides. Therefore, smart contract upgrades require changing the implementation contract while retaining the same storage structure, necessitating careful planning. 
+
+Another downside is that smart contracts often follow a gas metering model, where program execution is associated with a given unit and a marketplace is set up to pay for such an execution unit. This fee system is often very rigid, and some complex flows (like account abstraction, among others) have been developed to circumvent this problem.
+
+In contrast, parachains can create their own custom logic and state transition function (STF or runtime) thanks to the modularity provided by the [Polkadot-SDK](TODO:update-path){target=\_blank}. Consequently, logic executed within the parachain runtime can give developers a lot of flexibility when building applications on top of it.
```

Comment:
- If this link is intended to redirect to the "Introduction to Polkadot SDK" page, I think we can keep the `TODO:update-path` but it should be consistent, like: 
    "... provided by the Polkadot SDK, as stated in the 'Introduction to Polkadot SDK' page."
    
- If not, I think the link should be https://github.com/paritytech/polkadot-sdk/tree/master

Note: I believe the second option makes more sense in this context